cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaClusterCreator.java

```diff
@@ -113,6 +123,8 @@ public Future<KafkaCluster> prepareKafkaCluster(
                 .compose(kafka -> brokerRemovalCheck(kafkaCr, kafka))
                 .compose(kafka -> {
                     if (checkFailed() && tryToFixProblems)   {
+                        // saving scaling down blocked nodes, before they are reverted back
+                        this.scalingDownBlockedNodes.addAll(kafka.removedNodes());
```

Comment:
I see your point but this way it means to change a long chain of methods further down the `KafkaCluster.fromCrd` for passing that info, with all the consequences on updating tests as well. I will think more about it.

Reply:
From my understanding the `tryToFixProblems` boolean is also only set to true when you call the method from line 132, so perhaps it could work to have one method which is public that omits `tryToFixProblems` and `scalingDownBlockedNodes` and then a second method which the first one calls that is private and has those two arguments.

Reply:
I am not sure I have got what you meant but it looks like a refactoring not strictly related to the `scalingDownBlockedNodes`? I mean, taking this apart, does your suggestion still apply?